cmd/kaf/produce.go

```diff
@@ -17,28 +19,57 @@ import (
 var (
 	keyFlag         string
 	headerFlag      []string
-	numFlag         int
+	repeatFlag      int
 	partitionerFlag string
 	timestampFlag   string
 	partitionFlag   int32
+	bufferSizeFlag  int
+	inputModeFlag   string
 )
 
 func init() {
 	rootCmd.AddCommand(produceCmd)
 
 	produceCmd.Flags().StringVarP(&keyFlag, "key", "k", "", "Key for the record. Currently only strings are supported.")
 	produceCmd.Flags().StringArrayVarP(&headerFlag, "header", "H", []string{}, "Header in format <key>:<value>. May be used multiple times to add more headers.")
-	produceCmd.Flags().IntVarP(&numFlag, "num", "n", 1, "Number of records to send.")
+	produceCmd.Flags().IntVarP(&repeatFlag, "repeat", "n", 1, "Repeat records to send.")
 
 	produceCmd.Flags().StringSliceVar(&protoFiles, "proto-include", []string{}, "Path to proto files")
 	produceCmd.Flags().StringSliceVar(&protoExclude, "proto-exclude", []string{}, "Proto exclusions (path prefixes)")
 	produceCmd.Flags().StringVar(&protoType, "proto-type", "", "Fully qualified name of the proto message type. Example: com.test.SampleMessage")
 
 	produceCmd.Flags().StringVar(&keyProtoType, "key-proto-type", "", "Fully qualified name of the proto key type. Example: com.test.SampleMessage")
-	produceCmd.Flags().StringVar(&partitionerFlag, "partitioner", "", "Select partitioner: Default or jvm")
+	produceCmd.Flags().StringVar(&partitionerFlag, "partitioner", "", "Select partitioner: [jvm|rand|rr|hash]")
 	produceCmd.Flags().StringVar(&timestampFlag, "timestamp", "", "Select timestamp for record")
 	produceCmd.Flags().Int32VarP(&partitionFlag, "partition", "p", -1, "Partition to produce to")
 
+	produceCmd.Flags().IntVarP(&bufferSizeFlag, "buffer-size", "B", 0, "Buffer size for scanning input")
+	produceCmd.Flags().StringVarP(&inputModeFlag, "input-mode", "M", "byline", "Scanning input mode: [single|byline]")
```

Comment:
i don't really like to use shorthands (`-M` here) for more "complicated" or "advanced" flags.. i'd rather have no shorthand here.

also: i'd prefer to see `by-line` instead of `byline`. i was confused a bit; "what's a byline" ? ;)
for "single", i don't think anybody understands what "single" means. it's more of a "full" mode, where the whole input is read at once.

since i'm not very satisfied with this flag, and i don't know better, i googled a bit to find out what other unix tools do.

for example, grep has a flag `--line-buffered ` for this purpose.
```
      --line-buffered       flush output on every line
```
what do you think? it's what we want to do here, right? if it makes sense, i'd go with this.

Reply:
I think the flag  `--line-length-limit`  is more readable than `--line-buffer`  

`byline|single` is from https://linux.die.net/man/8/ngrep

`line|full` also sounds good to me

Reply:
i'd go with `line|full` :)